engine/games/pikachu_minigame.asm

```diff
@@ -0,0 +1,1449 @@
+	ld [wPikachuMinigameTimeFrames], a
+	ld [wPikachuMinigameTimeSeconds], a
+	ld [wc613], a
+	ld [wc614], a
+	ld [wPikachuMinigameNoteTimer], a
+	ld [wPikachuMinigameSpawnTypeIndex], a
+	ld [wPikachuMinigameSpawnDataIndex], a
+	ld [wPikachuMinigameRedrawTimer], a
+	ld [wPikachuMinigameColumnFlags], a
+	ld [wPikachuMinigameSavedColumnPointer], a
+	ld [wPikachuMinigameSavedColumnPointer + 1], a
+	ld [wPikachuMinigameRepeatColumnCounter], a
+	ld [wPikachuMinigameRepeatColumnCounter2], a
+
+	ld a, LOW(PikachuMiniGame_Columns)
+	ld [wPikachuMinigameColumnPointer], a
+	ld a, HIGH(PikachuMiniGame_Columns)
+	ld [wPikachuMinigameColumnPointer + 1], a
+
+; Set status bar position
+	ld a, $7c
+	ldh [hWY], a
+
+	ld a, $e3
+	ldh [rLCDC], a
+
+	ld a, [wSGB]
+	and a
+	jr nz, .not_sgb
+
+	ld a, %10010100
+	ldh [rBGP], a
+	ld a, %11100100
+	ldh [rOBP0], a
+	jr .load_pikachu
+
+.not_sgb
+; Normal palette if on GB / GBC
+	ld a, %11100100
+	ldh [rBGP], a
+	ldh [rOBP0], a
+
+.load_pikachu
+	depixel 14, 11
+	ld a, SPRITE_ANIM_INDEX_MINIGAME_PIKACHU
+	call InitSpriteAnimStruct
+
+; Save pointer to the newly initialized Pikachu object
+	ld a, c
+	ld [wPikachuMinigamePikachuObjectPointer], a
+	ld a, b
+	ld [wPikachuMinigamePikachuObjectPointer + 1], a
+
+; load Pikachu's tail object
+	depixel 14, 11
+	ld a, SPRITE_ANIM_INDEX_MINIGAME_PIKACHU_TAIL
+	call InitSpriteAnimStruct
+
+	ld a, c
+	ld [wPikachuMinigamePikachuTailObjectPointer], a
+	ld a, b
+	ld [wPikachuMinigamePikachuTailObjectPointer + 1], a
+	ret
+
+PikachuMiniGame_ClearBothTilemaps:
+	ld hl, vBGMap1
+	ld bc, SCREEN_WIDTH * BG_MAP_HEIGHT
+.clear_bgmap
+	ld [hl], 0
+	inc hl
+	dec bc
+	ld a, b
+	or c
+	jr nz, .clear_bgmap
+
+	ld hl, wTileMap
+	ld bc, SCREEN_WIDTH * SCREEN_HEIGHT
+.clear_tilemap
+	ld [hl], 0
+	inc hl
+	dec bc
+	ld a, b
+	or c
+	jr nz, .clear_tilemap
+
+	ld a, 7
+	ldh [hWX], a
+	ret
+
+PikachuMiniGame_LoadFont:
+	ld hl, FontGFX
+	ld de, vFont tile $10
+	ld bc, (112 / 2) tiles
+	ld a, BANK(FontGFX)
+	call FarCopyDataDouble
+
+	ld hl, FontGFX tile $39
+	ld de, vChars2 tile $32
+	ld bc, ( (14 / 2) + 1) tiles
+	ld a, BANK(FontGFX)
+	call FarCopyDataDouble
+	ret
+
+PikachuMiniGame_BlinkText:
+; Blink the window text according to the current X scroll position.
+; 00 - 7f = hide text
+; 80 - ff = show text
+
+	ldh a, [hSCX]
+	ld d, a
+	and $7f
+	ret nz
+	bit 7, d
+	jr nz, .DisplayText
+
+; clear dakutens
+	xor a
+	hlcoord 5, 0
+	ld [hl], a
+	hlcoord 13, 0
+	ld [hl], a
+
+; clear text
+	hlcoord 1, 1
+	ld c, 15
+.text_clear
+	ld [hli], a
+	dec c
+	jr nz, .text_clear
+	ret
+
+.DisplayText:
+	decoord 1, 1
+	ld hl, .text
+
+.render_text
+	ld a, [hli]
+	and a
+	jr z, .render_dakutens
+
+; Tiles are shifted so add $10 to each character
+	add $10
+	ld [de], a
+	inc de
+	jr .render_text
+
+.render_dakutens
+; Render dakuten marks separately
+	ld a, "ﾞ" + $10
+
+	hlcoord 5, 0
+	ld [hl], a
+
+	hlcoord 13, 0
+	ld [hl], a
+
+	ret
+
+.text
+	db "スタートホタン▶タイトルかめん"
+	;  "スタートボタン▶タイトルがめん"
```

Comment:
No need for this comment

Reply:
Since the dakuten marks were placed independently of the string, I felt like adding that comment to show the proper text - but removing as requested